nixos/modules/config/zram.nix

```diff
@@ -24,10 +24,21 @@ in
         type = lib.types.bool;
         description = ''
           Enable in-memory compressed devices and swap space provided by the zram
-          kernel module.
+          kernel module. This is useful if you prefer to avoid setting up a
+          swap partition or swap file stored on a storage device.
+
           See [
             https://www.kernel.org/doc/Documentation/blockdev/zram.txt
           ](https://www.kernel.org/doc/Documentation/blockdev/zram.txt).
+
+          If you intend to use swap on-disk as well, then instead of using zramSwap,
+          consider enabling
+          [zswap](https://www.kernel.org/doc/html/latest/admin-guide/mm/zswap.html).
+          Zswap applies more suitable heuristics in this case when using disk swap.
+
+          Zswap can be enabled using `boot.kernelParams = [ "zswap.enabled=1" ];`.
```

Comment:
Maybe this is a good time to introduce a zswap module then? We could then direct users to the right options, or even add a warning.

Reply:
I think it would make sense to do this in a second PR.

Reply:
What I really don't like is that we have information in the zram docs that's not very relevant to zram, and we could have automation take care of this instead of relying on users to read things (which they never do).